src/stcal/ramp_fitting/src/slope_fitter.c

```diff
@@ -899,23 +927,29 @@ clean_ramp_data(
     struct simple_ll_node * current;
     struct simple_ll_node * next;
 
-    if (NULL == rd->segs) {
-        return; /* Nothing to do. */
-    }
-
-    /*
-     *  For each pixel, check to see if there is any allocated
-     *  memory for the linked list of ramp segments and free them.
-     */
-    for (idx=0; idx < rd->cube_sz; ++idx) {
-        current = rd->segs[idx];
-        if (current) {
-            next = current->flink;
-            SET_FREE(current);
-            current = next;
+    Py_XDECREF(rd->data);
+    Py_XDECREF(rd->groupdq);
+    Py_XDECREF(rd->err);
+    Py_XDECREF(rd->pixeldq);
+    Py_XDECREF(rd->zframe);
+    Py_XDECREF(rd->dcurrent);
```

Comment:
Should this contain `Py_XDECREF`s for `gain` and `rnoise`?

Reply:
This was part of the additional testing I wanted to do.  These variables get set at:

https://github.com/kmacdonald-stsci/stcal/blob/2df21c4fb36e1d302717771fbc4b8878781fc31a/src/stcal/ramp_fitting/src/slope_fitter.c#L1772

This doesn't appear to increment the reference count.  If I include `Py_XDECREF` for `gain` and `rnoise`, this causes problems when returning from ramp fitting.  When these variables are used during the read noise recalculation for CHARGELOSS those objects aren't available anymore.